src/puget/dispatch.cljc

```diff
@@ -19,8 +19,9 @@
    {:pre [(sequential? dispatchers)]}
    (let [candidates (remove nil? dispatchers)]
      (when (empty? candidates)
-       (throw (IllegalArgumentException.
-                "chained-lookup must be provided at least one dispatch function to try.")))
+       #?(:clj (throw (IllegalArgumentException.
```

Comment:
For exceptions, I think a better route is to use the platform-agnostic `ex-info` constructor.